docs/ops/detection/GenerateProposals_9.md

```diff
@@ -0,0 +1,147 @@
+# GenerateProposals {#openvino_docs_ops_detection_GenerateProposals_9}
+
+**Versioned name**: *GenerateProposals-9*
+
+**Category**: *Object detection*
+
+**Short description**: The *GenerateProposals* operation proposes ROIs and their scores
+based on input data for each image in the batch.
+
+**Detailed description**: The operation performs the following steps for each image:
+
+1.  Transposes and reshapes predicted bounding boxes deltas and scores to get them into the same dimension order as the
+anchors.
+2.  Transforms anchors and deltas into proposal bboxes and clips proposal bboxes to an image. The attribute *normalized*
+indicates whether the proposal bboxes are normalized or not.
+3.  Sorts all `(proposal, score)` pairs by score from highest to lowest; order of pairs with equal scores is undefined.
+4.  Takes top *pre_nms_count* proposals, if total number of proposals is less than *pre_nms_count* takes all proposals.
+5.  Removes predicted boxes with either height or width < *min_size*.
```

Comment:
In `ExperimentalDetectronGenerateProposalsSingleImage` this step goes at third stage. When it makes sense to do at the third step - any FW requires this? Maybe we need to introduce a separate flag. So we will manage to deprecate `ExperimentalDetectronGenerateProposalsSingleImage`. Please investigate this.

Reply:
Do you mean a flag to indicate whether the min_size filter should put before partial sort or after partial sort?
But I see no other framework will put min_size filter before partial sort. They all do min_size filtering after sort like this spec describes.
references:
Caffe2: https://github.com/pytorch/pytorch/blob/4dae85c38517fd066a67901f660ca3bc08da7446/caffe2/operators/generate_proposals_op.cc#L245
Paddle: https://github.com/PaddlePaddle/Paddle/blob/8803f6bb1e390a56c1cb2f0b7dcb754d872f66ab/paddle/fluid/operators/detection/generate_proposals_v2_op.cc#L214
tensorflow: https://github.com/tensorflow/tensorflow/blob/25e649ac3688655547da998eba2715cf70b3e5c9/tensorflow/core/kernels/image/generate_box_proposals_op.cu.cc#L129

I think it may be not necessary to be compatible with ExperimentalDetectronGenerateProposalsSingleImage at this point. What do you think?